OAuth: Feature toggle for access token expiration check and docs (#58179)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md

**Note:** Verify that the Grafana [root_url]({{< relref "../../../configure-grafana/#root-url" >}}) is set in your Azure Application Redirect URLs.

> ### Access token expiration check and configure refresh token

Reviewer: I would simply go with ### Configure refresh token

The reason: You address the access token expiration check in the intro text that follows and this section is really about the task of configuring a refresh token.

Author: Done, thanks!

> Available in Grafana v9.3 and later versions.

> **Note:** This feature is behind the `accessTokenExpirationCheck` feature toggle.

> Grafana verifies that the User's access token is not expired in case the User is logged in using an OAuth provider and it tries to obtain a new access token by using the provided refresh token (if any) if the access token is expired.

Reviewer 2: nit (here and below): I'd break this sentence up into two parts to make it easier to read. Something like:
```suggestion
Grafana verifies that the User's access token is not expired in case the User is logged in using an OAuth provider. When the access token expires, Grafana tries to obtain a new access token by using the provided refresh token (if any).
```

> A refresh token is a special token that is used to get new access tokens without having to ask the user to log in again. Without a refresh token the User gets logged out from the system after the access token is expired.

Reviewer: Unable to leave a suggestion for some reason. Maybe use this wording?

"Grafana uses a refresh token to obtain a new access token without requiring the user to log in again. If a refresh token doesn't exist, Grafana logs the user out of the system after the access token has expired."

> 1. Extend the `scopes` in `[auth.azuread]` with `offline_access`

Reviewer 2: thought: maybe worth prefixing this with something like "To enable refresh token for AzureAD:"?

Reviewer: It appears that this task has just a single step (which can happen, and that's ok) - in this case, you do not need to include a step number (because there is no step 2).

Author: Thanks, it's been reworded.

> 1. Extend the `scopes` in `[auth.azuread]` with `offline_access`

Reviewer: ```suggestion
To enable a refresh token for AzureAD, extend the `scopes` in `[auth.azuread]` with `offline_access`.
```
